javalib/src/main/scala/java/util/concurrent/ConcurrentSkipListSet.scala

```diff
@@ -0,0 +1,404 @@
+package java.util.concurrent
+
+import java.util._
+import java.lang.Comparable
+import scala.collection.mutable
+import scala.math.Ordering
+
+import scala.collection.JavaConversions._
+
+class ConcurrentSkipListSet[E >: Null] protected (ordering: Ordering[E], provided: Boolean) 
+    extends AbstractSet[E]
+    with NavigableSet[E]
+    with Cloneable
+    with Serializable { self =>
+
+  def this() =
+    this(ConcurrentSkipListSet.defaultOrdering[E], false)
+
+  def this(comparator: Comparator[E])  = 
+    this(Ordering.comparatorToOrdering(comparator), true)
+
+  def this(sortedSet: SortedSet[E]) = {
+  	this(Ordering.comparatorToOrdering(sortedSet.comparator()), true)
+    addAll(sortedSet)
+  }
+
+  def this(collection: java.util.Collection[E]) = {
+  	this(ConcurrentSkipListSet.defaultOrdering[E], false)
+    addAll(collection)
+  }
+
+  def this(comparator: Comparator[E], collection: Collection[E]) = {
+    this(Ordering.comparatorToOrdering(comparator), true)
+    addAll(collection)
+  }
+
+  implicit object BoxOrdering extends Ordering[Box[E]] {
+    def compare(a: Box[E], b:Box[E]): Int = ordering.compare(a.inner, b.inner)
+  }
+
+  private val inner: mutable.SortedSet[Box[E]] = new mutable.TreeSet[Box[E]]()
+
+  override def isEmpty(): Boolean =   
+    inner.isEmpty
+
+  def iterator(): Iterator[E] = {
```

Comment:
Is this documented anywhere? Or is it just an "accident" of the current implementation (which might change without notice)? The ScalaDoc doesn't seem to give any guarantee about that.

[The current implementation](https://github.com/scala/scala/blob/v2.11.6/src/library/scala/collection/mutable/TreeSet.scala#L105-L115) obviously satisfies it because it basically works with a var of an immutable red-black tree internally, and `iterator()` works on a snapshot of this var. But this is a private implementation detail, which is not guaranteed to be preserved.